app/models/concerns/smooch_newsletter.rb

```diff
@@ -77,8 +77,13 @@ def newsletter_content_changed?(newsletter, language, team_id)
 
     def newsletter_cron(newsletter)
       hour = newsletter['smooch_newsletter_time'].to_i
-      timezone = newsletter['smooch_newsletter_timezone'].to_s.upcase
-      # Mapping for timezones not supported by Ruby's DateTime
+      # If an offset is being passed, it's in the new format
+      if newsletter['smooch_newsletter_timezone'].match?(/\W\d\d:\d\d/)
+        timezone = newsletter['smooch_newsletter_timezone'].match(/\W\d\d:\d\d/)
+      else
```

Comment:
@caiosba I'm afraid that if I get rid of the conditional here and only look for the new format, there will be a situation where we try to parse a date with a null/undefined value for the timezone and there will be some kind of error thrown. This is mostly me not knowing how Ruby's `match` and `DateTime.parse` really work in failure cases.

Reply:
OK, makes sense, it doesn't hurt to have a fallback... better be safe.

Reply:
Let's just be sure that the `else` case is also covered in that unit test.